packages/data-lake-api/src/DataLakeApi.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import groupBy from 'lodash.groupby';
+
+import moment from 'moment';
+import { getSortByKey } from '@tupaia/utils';
+import { DataLakeAnalyticsFetchQuery } from './DataLakeAnalyticsFetchQuery';
+import { DataLakeEventsFetchQuery } from './DataLakeEventsFetchQuery';
+import { validateEventOptions, validateAnalyticsOptions } from './validation';
+import { sanitizeAnalyticsTableValue } from './sanitizeAnalyticsTableValue';
+import { sanitiseFetchDataOptions } from './sanitiseFetchDataOptions';
+import { DataLakeDatabase } from './DataLakeDatabase';
+
+const EVENT_DATE_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
+
+const buildEventDataValues = resultsForEvent =>
+  resultsForEvent.reduce(
+    (values, { dataElementCode, type, value }) => ({
+      ...values,
+      [dataElementCode]: sanitizeAnalyticsTableValue(value, type),
+    }),
+    {},
+  );
+
+let dataLakeDatabase;
+
+const getDatabase = () => {
+  if (!dataLakeDatabase) {
+    dataLakeDatabase = new DataLakeDatabase();
+  }
+  return dataLakeDatabase;
+};
```

Comment:
Used this singleton pattern so that there's just a single connection to the data-lake database. Not entirely happy with it, but there is some trickiness around managing a long-lived connection when the whole data-broker->api chain gets re-created each data fetch call...

Reply:
Seems like the singleton decision should be made by whoever instantiates a DataBroker, and then databroker should create services once, which create apis once. So I would move this to the constructor.

Reply:
Ok sounds good